tests/unit/legacy/api/xmlrpc/test_xmlrpc.py

```diff
@@ -114,6 +114,23 @@ def view(context, request):
 
 
 class TestSearch:
+    def test_error_when_disabled(self, pyramid_request, metrics, monkeypatch):
+        monkeypatch.setattr(
+            pyramid_request.registry,
+            "settings",
+            {"warehouse.xmlrpc.search.enabled": False},
+        )
+        with pytest.raises(xmlrpc.XMLRPCWrappedError) as exc:
+            xmlrpc.search(pyramid_request, {"name": "foo", "summary": ["one", "two"]})
+
+        assert exc.value.faultString == (
+            "RuntimeError: This API has been deprecated due to unmanageable load. "
```

Comment:
hmmm, this message seems _perfect_ for the final shut off, but in the meantime I might. suggest

```RuntimeError: This API has been temporarily disabled to to unmanageable load and will be deprecated in near future. Please use the Simple or JSON API instead.```